src/gl/fpe.h

```diff
@@ -187,7 +188,8 @@ typedef struct fpe_state_s {
     unsigned int fragment_prg_enable:1;  // if fragment program is enabled
     uint16_t     vertex_prg_id;          // Id of vertex program currently binded (0 most of the time), 16bits is more than enough...
     uint16_t     fragment_prg_id;        // Id of fragment program currently binded (0 most of the time)
-}__attribute__((packed)) fpe_state_t;
```

Comment:
the `__attribute__((packed))` is GCC only?

Reply:
> 
> 
> the `__attribute__((packed))` is GCC only?

__attribute__ is gcc-specific keyword (supported also by clang).
To change the packing, some compilers have constructions that are similar in semantics, but the version with
#pragma pack(1) 
is universal (last ~15 years :).

In a few days I expect to propose a separate commit, where all __attribute__(visibility) will be replaced by a macro
(close to how you did it in the STUB macro - since the semantics of describing exports, for example, on a windows
platform is very different.
Moreover, __attribute__(visibility) gcc (in the mingw incarnation) and clang, on Windows perceive, but... do not use
(they say it is elf-specific).

Reply:
For windows, I thing the Visibilty attribute needs to be change to "decl export"

Reply:
And yes, "alias" are probably a linux/elf only stuff, so an alternative solution will need to be find (creating a simple function that call the gles_ one and return the correct argument type kind of macro...)

Reply:
> 
> 
> For windows, I thing the Visibilty attribute needs to be change to "decl export"

__declspec(dllexport) must be BEFORE function declaration, and __attribute__(((visibility - AFTER
for one function we can write
---
#ifdef _WIN32
__declspec(dllexport)
#endif 
rettype CC func(args)
#ifndef _WIN32
__attribute__((visibilty("default")))
#endif
;
---
;-)
but for multiple function....

Reply:
the AliasExport macro has to be redone probably.

Reply:
> 
> 
> the AliasExport macro has to be redone probably.

...if redesign it to form
AliasExport(internalname, externalname)
yes. 
But at the same time you have to rule all the places where it is used.
I can do so, but the option with a separate def file does not require editing in a heap of place (only 2-3). True it (this file) will have to be maintained (when you add functions).
What do you like better? '-)
---
sorry, but my english is very poor

Reply:
More like AliasExport(internalname, externalname, returntype, args, callingargs).
Look at this for example: https://github.com/ptitSeb/box86/blob/master/src/tools/gtkclass.c#L117

Reply:
>
> More like AliasExport(internalname, externalname, returntype, args, callingargs).

I mentioned this kind of prince - "close to how you did it in the STUB macro" ;-)

Ok, so I'll try to implement it (tomorrow-after-tomorrow).